org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -75,9 +78,10 @@
 
     private IProviderContext providerContext;
     private VariableRequestHandler variableRequestHandler;
-
     private IDebugAdapterContext debugContext = null;
     private Map<Command, List<IDebugRequestHandler>> requestHandlers = null;
+    
+    private static final IThreadResourceDisposer ON_DEMAND_THREAD_RESOURCE_DISPOSER = new OnDemandThreadResourceDisposer();
```

Comment:
put the disposer in debugcontext rather than change the signature of registerHandler

Reply:
each handler could register different disposers